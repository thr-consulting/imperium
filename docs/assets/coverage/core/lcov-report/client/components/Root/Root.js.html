<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/Root/Root.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">client/components/Root</a> Root.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
/* eslint-disable react/prefer-stateless-function */
&nbsp;
import React, {Component} from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import {RouteDirector} from '@thx/router';
&nbsp;
import routeDefaults from 'routeDefaults';
import './root.css';
&nbsp;
type Props = {
	store: Object,
	routes: Object[],
	render: () =&gt; void,
};
&nbsp;
/**
 * The root component
 */
export default class Root extends Component&lt;Props&gt; {
	props: Props;
&nbsp;
	render() {
		const {store, routes, render} = this.props;
		const inner = () =&gt; (
			&lt;Router&gt;
				&lt;RouteDirector routes={routes} defaults={routeDefaults}/&gt;
			&lt;/Router&gt;
		);
		return (
			&lt;Provider store={store}&gt;
				&lt;div&gt;
					{render(inner)}
				&lt;/div&gt;
			&lt;/Provider&gt;
		);
	}
&nbsp;
	/*
	render() {
		const {store, apolloClient, routes} = this.props;
		return (
			&lt;ApolloProvider client={apolloClient}&gt;
				&lt;Provider store={store}&gt;
					&lt;div&gt;
						&lt;NotificationSystem&gt;
							&lt;DialogSystem&gt;
								&lt;Router&gt;
									&lt;RouteDirector routes={routes} defaults={routeDefaults}/&gt;
								&lt;/Router&gt;
							&lt;/DialogSystem&gt;
						&lt;/NotificationSystem&gt;
					&lt;/div&gt;
				&lt;/Provider&gt;
			&lt;/ApolloProvider&gt;
		);
	}
	*/
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 12:01:38 GMT-0600 (CST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
