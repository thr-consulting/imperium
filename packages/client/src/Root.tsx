import debug from 'debug';
import React from 'react';
import {IImperiumClient, RootProps} from './types';

const d = debug('imperium.client.Root');

interface Props {
	render: (props: RootProps) => React.ReactNode;
	imperiumClient: IImperiumClient;
}

export default function Root(props: Props) {
	return props.render(props);
}
