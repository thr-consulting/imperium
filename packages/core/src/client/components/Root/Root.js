// @flow
/* eslint-disable react/prefer-stateless-function */

import React, {Component} from 'react';
import debug from 'debug';
import {BrowserRouter as Router} from 'react-router-dom';
import {hot} from 'react-hot-loader/root';
import routeDefaults from 'routeDefaults';
import RouteDirector from '../RouteDirector';
import './root.css';

const d = debug('imperium.core.Root');

type Props = {
	routes: Object[],
	render: () => void,
	startupData?: Object,
};

/**
 * The root component
 */
class Root extends Component<Props> {
	props: Props;

	render() {
		d('Rendering Root component');
		const {routes, render, startupData} = this.props;

		const Child = props => (
			<RouteDirector routes={routes} defaults={routeDefaults} {...props}/>
		);

		return (
			<Router>
				{render({Child, ...startupData})}
			</Router>
		);
	}
}

export default hot(Root);
