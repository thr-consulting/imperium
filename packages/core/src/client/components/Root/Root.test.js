import React from 'react';
import {shallow} from 'enzyme';
import Root from './Root';

function checkPermissions() {
	return true;
}

describe('Root', () => {
	it('should render without throwing an error', () => {
		const render = jest.fn().mockImplementation(inner => inner({checkPermissions}));
		const routes = {MyRoute: 'testroute'};
		const store = {
			subscribe: jest.fn(),
			dispatch: jest.fn(),
			getState: jest.fn(),
		};

		expect(shallow(
			<Root render={render} store={store} routes={routes}/>
		)).toMatchSnapshot();
	});
});
