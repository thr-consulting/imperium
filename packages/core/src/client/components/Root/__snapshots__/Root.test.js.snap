// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should render without throwing an error 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <div>
    <BrowserRouter>
      <Component
        checkPermissions={[Function]}
        defaults="test-routeDefaults-stub"
        routes={
          Object {
            "MyRoute": "testroute",
          }
        }
      />
    </BrowserRouter>
  </div>
</Provider>
`;
